client/dwh_migration_client/config.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Parses the config file into a Config object."""
+
+import logging
+from dataclasses import asdict, dataclass
+from pprint import pformat
+from typing import List, Optional
+
+import yaml
+from marshmallow import Schema, ValidationError, fields, post_load
```

Comment:
Not surprisingly, this is also my first time learning what magic this package offers. And the change LGTM after ~10 mins of Google search on it.

A general/naive question,  what is the rule of thumb when we should consider to add a new package to replace some existing logics that are implemented in the old fashioned way? A new open-source package offers some level of improvements (otherwise it shouldn't exist in the first place) but it also raises the bars of maintenance/readability because we are asking readers to spend another ~10min learning this package.  And I think it also raises the risk of compatibility issues due to a new dependency added to this tool.

I am not saying we shouldn't add this change, just curious about the "best practice" in the open source Python  world.